manifests/init.pp

```diff
@@ -1,9 +1,18 @@
+# @summary module to configure a systemd timer + script to backup postgres, mysql and ldap dataabases
+#
+# @param pigz_cpu_cores number of cpu cores used for backup compression
+# @param destination directory where backups will be stored (with one subdirectory per backup day)
+# @param interval how often the systemd timer will trigger the dump script. This will be used for the OnUnitInactiveSec option in the timer
+# @param backuphistory number of days to keep the backup history
+# @param cleanup_empty_backup_dirs enable/disable a post command to cleanup empty directories with $destination. They can occour if the system run out of free diskspace
+#
+# @author Tim Meusel <tim@bastelfreak.de>
 class dbbackup (
-  Integer $pigz_cpu_cores             = $facts['processors']['count'],
+  Integer[1] $pigz_cpu_cores          = $facts['processors']['count'],
   Stdlib::Absolutepath $destination   = '/mnt/dumps',
-  String $interval                    = '1h',
+  String[2] $interval                 = '1h',
```

Comment:
What about `'30m'` or `'12h'`?  Are these not valid?

Reply:
Ignore me.  Need more sleep!

Reply:
Those are accepted. `2` is the min length, not the exact length. If it was `String[2, 2]` it would be a problem.